src/tests/Loader/classloader/ExtendedLayout/ExtendedLayoutTypes.il

```diff
@@ -0,0 +1,160 @@
+
+    .field public int32 a
+}
+
+// CStruct types below (can be moved to C# once we have ExtendedLayout support in the C# compiler)
+
+.class flags(0x18) public value beforefieldinit CStructBlittablePrimitiveFields
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public int32 a
+    .field public float32 b
+    .field public uint8 c
+}
+
+.class flags(0x18) public value beforefieldinit CStructNonBlittablePrimitiveFields
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public bool b
+    .field public char c
+}
+
+.class flags(0x18) public value beforefieldinit CStructWithReferenceFields
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public class [System.Runtime]System.String a
+}
+
+.class flags(0x18) public value beforefieldinit CStructWithMixedFields
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public int32 a
+    .field public class [System.Runtime]System.String b
+}
+
+.class flags(0x18) public value beforefieldinit NestedCStructType
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+    .field public int32 x
+}
+
+.class flags(0x18) public value beforefieldinit CStructCustomCStructField
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public valuetype NestedCStructType y
+}
+
+.class public sequential value beforefieldinit NestedSequentialType
+{
+    .field public int32 x
+}
+
+.class flags(0x18) public value beforefieldinit CStructCustomSeqStructField
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public valuetype NestedSequentialType y
+}
+
+.class public auto value beforefieldinit NestedAutoLayoutType
+{
+    .field public int32 x
+}
+
+.class flags(0x18) public value beforefieldinit CStructCustomAutoStructField
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+
+    .field public valuetype NestedAutoLayoutType y
+}
+
+.class flags(0x18) public value beforefieldinit EmptyCStruct
+{
+    .custom instance void [System.Runtime]System.Runtime.InteropServices.ExtendedLayoutAttribute::.ctor(int32) = {
+        int32(0x00000000) // CStruct
+    }
+}
```

Comment:
Technically an empty struct is invalid in C. This is an annoyance because gcc and clang returns a size of 0 and MSVC actually errors out. In C++ an empty struct is valid, but is required to return a size of 1. I'm not sure we want to permit empty for a type that explicitly states "CStruct".

Reply:
I'll block empty structs. By the same logic as the "pack" case, if people want this they can use `LayoutKind.Sequential`.